backend/src/api/PostController.ts

```diff
@@ -126,11 +126,21 @@ export default class PostController {
                 return response.error('error', 'Unknown error', 500);
             }
 
+            const restrictions = await this.userManager.getUserRestrictions(userId);
+
             const {posts: [post], users} = await this.enricher.enrichRawPosts([rawPost]);
+            if (!restrictions.canEditOwnContent) {
+                post.canEdit = false;
+            }
 
             let comments: CommentEntity[] = [];
             if (!noComments) {
                 const rawComments = await this.postManager.getPostComments(postId, userId, format);
+
+                if (!restrictions.canEditOwnContent) {
+                    rawComments.forEach(comment => comment.canEdit = false);
```

Comment:
`postManager.getPostComments` is already iterating comments, it might be more efficient to pass `canEdit` as an argument to this method and set this value there to avoid additional iteration

Reply:
Iteration is cheap compared to DB and and json serialization. I wouldn't worry about that (don't think it's worth adding extra responsibility to `getPostComments`)